frontend/src/views/WheresMyGene/components/Filters/components/Organism/index.tsx

```diff
@@ -45,14 +58,16 @@ export default function Organism({ isLoading }: Props): JSX.Element {
   useEffect(() => {
     if (!organisms || !dispatch || selectedOrganismId) return;
 
+    setAvailableOrganisms(organisms);
```

Comment:
I'm thinking it might be worth extracting a React Query to get available organisms by leveraging `usePrimaryFilterDimensions()`, since React Query is designed for getting entities from remote states, which is what we're doing here!

E.g.,

```ts
export function useAvailableOrganisms() {
  const { data, isLoading } = usePrimaryFilterDimensions();

  if (isLoading) {
    return { isLoading, data: null };
  }

  return {
    isLoading,
    data: organisms.filter((organism: IOrganism) =>
      TEMP_ALLOW_NAME_LIST.includes(organism.name)
    ),
  }
}
```

And then all call sites that need `availableOrganisms` can just use this new query hook:

```ts
export default function Organism(...) {
  const { data: organisms, isLoading } = useAvailableOrganisms();
  ...
}
```

Reply:
I did find a comment I wrote a year ago about the same pattern used on `availableFilters`

```ts
// (thuang): We only update available filters when API call is done,
// otherwise when `useFilterDimensions()` is still loading, its filters
// will temporarily be empty, and thus resetting the selected filter values
```

I'm hoping the comment doesn't apply to organisms because I don't think we refetch organisms as users select more filters. But if we do, then maybe leave the current implementation as is is fine and we just add a comment to explain why we can't extract it as a query hook 🙏 

Thank you!